### PR TITLE
add ability to disable one of the connext implementations

### PR DESCRIPTION
**Pull request by wjwwood**
Connects ros2/ros2#147

### CODE REVIEW

File: ros2_batch_job/linux_batch/__init__.py [hunk 1]
```diff
@@ -63,6 +63,13 @@ def setup_env(self):
         def with_vendors(cmd, **kwargs):
             # Ensure shell is on since we're using &&
             kwargs['shell'] = True
+            # Extend the environment if needed.
+            env = kwargs.get('env', {})
+            if self.args.disable_connext_static:
+                env["CONNEXT_STATIC_DISABLE"] = '1'
+            if self.args.disable_connext_dynamic:
+                env["CONNEXT_DYNAMIC_DISABLE"] = '1'
+            kwargs['env'] = dict(env)
```
Comment (dirk-thomas) [inline ros2_batch_job/linux_batch/__init__.py]: This block looks a bit complicated to me. At the end it set `env` anyway to be a dict. Why not do the following instead:

    env = kwargs.setdefault('env' ,{})
    if self.args.disable_connext_static:
        env["CONNEXT_STATIC_DISABLE"] = '1'
    if self.args.disable_connext_dynamic:
        env["CONNEXT_DYNAMIC_DISABLE"] = '1'

Comment (dirk-thomas) [inline ros2_batch_job/linux_batch/__init__.py]: The question is does passing an empty env dict change the current behavior? Does it need to be initialized with `os.environ`?

Comment (wjwwood) [inline ros2_batch_job/linux_batch/__init__.py]: I think it should be seeded with `os.environ`, 37b7893

File: ros2_batch_job/linux_batch/__init__.py [hunk 2]
```diff
@@ -63,6 +63,12 @@ def setup_env(self):
         def with_vendors(cmd, **kwargs):
             # Ensure shell is on since we're using &&
             kwargs['shell'] = True
+            # Extend the environment if needed.
+            env = kwargs.setdefault('env', dict(os.environ))
+            if self.args.disable_connext_static:
+                env["CONNEXT_STATIC_DISABLE"] = '1'
```
Comment (dirk-thomas) [inline ros2_batch_job/linux_batch/__init__.py]: Please use single quotes.

Same below three more times.

**Comment by dirk-thomas** [comment]
lgtm

**Pull request merged**